birdhouse/scripts/read-configs.include.sh

```diff
@@ -16,15 +16,26 @@
 #
 #  # Source the script providing function read_configs.
 #  # read_configs uses COMPOSE_DIR to find default.env and env.local.
-#  . $COMPOSE_DIR/read-configs.include.sh
+#  . ${COMPOSE_DIR}/scripts/read-configs.include.sh
 #
 #  # Call function read_configs to read the various config files in the
 #  # appropriate order and process delayed eval vars properly.
 #  read_configs
 
 
+THIS_FILE="$(readlink -f "$0" || realpath "$0")"
+THIS_DIR="$(dirname "$THIS_FILE")"
+
+if [ -f "${THIS_DIR}/logging.include.sh" ]; then
+    . "${THIS_DIR}/logging.include.sh"
```

Comment:
> `COMPOSE_DIR` is not defined at that point.

Yes I understood that.  That is why I wrote "force call `discover_compose_dir` before sourcing `logging.include.sh`".

Basically the behavior will change to: if `read-configs.include.sh` is sourced, even if no other functions is called, it will have to ensure `COMPOSE_DIR` is set or display an error that `COMPOSE_DIR` not set, or set to an invalid location, and many features will not work.

Like I said earlier, any scripts, even external scripts can source `read-configs.include.sh`.  Therefore `THIS_FILE="$(readlink -f "$0" || realpath "$0")"` will return the location of the external script, not the location of `read-configs.include.sh` so you can not reliably use that to find `logging.include.sh`.

This is something I had in mind inside `read-configs.include.sh`:
```sh
discover_compose_dir
if [ ! -f "${COMPOSE_DIR}/pavics-compose.sh" ]; then
  echo "***CRITICAL: '${COMPOSE_DIR}/pavics-compose.sh' not found, please set variable 'COMPOSE_DIR' to a valid location.  Many features depend on this variable." 1>&2
fi

if [ -f "${COMPOSE_DIR}/scripts/logging.include.sh" ]; then
    . "${COMPOSE_DIR}/scripts/logging.include.sh"
fi
```

Reply:
I'm confused about our goals now:

@tlvu you would like to able to source the `read-configs.include.sh` from external scripts as discussed here (https://github.com/bird-house/birdhouse-deploy/pull/415#discussion_r1430715621) but you also want to raise a critical warning if the compose dir doesn't exist (which it may not if  `read-configs.include.sh` is sourced from an external script).

This critical warning says to me that sourcing `read-configs.include.sh` from anywhere outside of the locations expected by the `discover_compose_dir` function is not recommended. Is that your intention?

Reply:
> This critical warning says to me that sourcing `read-configs.include.sh` from anywhere outside of the locations expected by the `discover_compose_dir` function is not recommended. Is that your intention?

@mishaschwartz 

That is not my intention.  My intention has not changed, external script are encouraged to source `read-configs.include.sh`.  However, they should set `COMPOSE_DIR` themselves if it can not be autodetected by `discover_compose_dir`.

See existing warning here https://github.com/bird-house/birdhouse-deploy/blob/7e86368160ded7b4adc3887b922a67014770cdaa/birdhouse/read-configs.include.sh#L26-L29

Reply:
> "force call discover_compose_dir before sourcing logging.include.sh".

Do we want to move `discover_compose_dir` definition upward then? 
Currently, the function is not yet defined when the line to source logging gets called.

Reply:
Alternatively, do we want to improve how `${THIS_DIR}` gets resolved? I have seen alternatives using `$_` to handle sourced file location. The only tricky part is that different shells set it differently, so we need a few more if conditions to handle zsh/bash/sh.

Reply:
> Do we want to move `discover_compose_dir` definition upward then?

Yes, or rather you mean add the new code below the existing function `discover_compose_dir` definition  :D

Reply:
done